consensus/hotstuff_replica.go

```diff
@@ -195,20 +208,40 @@ func (m *ConsensusModule) validateProposal(msg *typesCons.HotstuffMessage) error
 	}
 
 	// Safety: check the hash of the locked QC
-	// TODO(olshansky): Extend implementation to adopt `ExtendsFrom` as described in the Hotstuff whitepaper.
-	if protoHash(lockedQC.Block) == protoHash(justifyQC.Block) { // && lockedQC.Block.ExtendsFrom(justifyQC.Block)
+	// The equivalent of `lockedQC.Block.ExtendsFrom(justifyQC.Block)` in the hotstuff whitepaper is done in `applyBlock` below.
+	if protoHash(lockedQC.GetBlock()) == protoHash(justifyQC.Block) {
 		m.nodeLog(typesCons.ProposalBlockExtends)
 		return nil
 	}
 
-	// Liveness: node is locked on a QC from the past. [TODO]: Do we want to set `m.LockedQC = nil` here or something else?
+	// Liveness: node is locked on a QC from the past.
+	// DISCUSS: Where should additional logic be added to unlock the node?
 	if justifyQC.Height > lockedQC.Height || (justifyQC.Height == lockedQC.Height && justifyQC.Round > lockedQC.Round) {
```

Comment:
noop: just an idea

What if we had something like:

```go
if nodeIsLockedOnPastQC(justifyQC, lockedQC) {
```

and then decompose `nodeIsLockedOnPastQC` further into functions that return the logical pieces we need. I think this is valid especially if we are considering the idea of adding additional logic here that could make this `if` very long and hard to understand IMHO.

Reply:
Checkout https://github.com/pokt-network/pocket/pull/250/commits/a8e5f3dae04d54da3745c502e14ab27801578f69